### PR TITLE
Add Host Memory Alignment Option

### PR DESCRIPTION
**Pull request by MrBurmark**
# Add Host Memory Alignment Option

This adds a command line option --align that lets you change the alignment of host memory allocations.

- This PR is a feature
- It does the following (modify list as needed):
  - Adds user settable memory alignment at the request of Corbin and me

### CODE REVIEW

File: src/common/RunParams.cpp [hunk 1]
```diff
@@ -316,6 +318,33 @@ void RunParams::parseCommandLineOptions(int argc, char** argv)
         input_state = BadInput;
       }
 
+    } else if ( opt == std::string("--align") ||
```
Comment (rhornung67) [inline src/common/RunParams.cpp:321]: Typically, we do a long form with '--' prefix and a short form with '-' prefix. Is that the intent here; i.e., '---align' is the short form? If so, I think it should have only one dash.

Also, is 'data' alignment a better name than 'memory' alignment?

File: src/common/RunParams.hpp [hunk 2]
```diff
@@ -125,6 +125,8 @@ class RunParams {
 
   double getSizeFactor() const { return size_factor; }
 
+  size_t getMemoryAlignment() const { return memory_alignment; }
```
Comment (rhornung67) [inline src/common/RunParams.hpp:128]: Data alignment instead of Memory Alignment?

File: src/common/RunParams.cpp (hunk 1 above)
Comment (MrBurmark) [reply src/common/RunParams.cpp:321]: that's reasonable, I'll change that

File: src/common/RunParams.hpp (hunk 2 above)
Comment (MrBurmark) [reply src/common/RunParams.hpp:128]: That seems to be what we're using for RAJA so I'll make that change for consistency.

**Review by rhornung67 — APPROVED** [review]

**Pull request merged**